### ISSUE TITLE
Marathon Leaderboard displays Max Rating(SRM, MM) should display MM Rating

### ISSUE THREAD

**Issue by harshitmehta15** [topcoder-platform/community-app#4947]
The user rating displayed on MM leaderboard is max(SRMrating, MMrating). Should be MM rating.

**Comment by harshitmehta15** [comment]
@Oanh-and-only-Oanh 
I think this got changed after v5 updates, can we revert this back?

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30142336 has been created for this ticket.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by Oanh-and-only-Oanh** [comment]
Submit in 8 hours and PR to:

https://github.com/topcoder-platform/community-app/tree/milestone-20200924
https://github.com/topcoder-platform/topcoder-react-lib/tree/milestone-20200924

run npm run test before PR

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30142336 has been updated - it has been assigned to cagdas001.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by Oanh-and-only-Oanh** [comment]
@cagdas001, please resolve this ticket first https://github.com/topcoder-platform/community-app/issues/4875. The other is high priority.

**Comment by cagdas001** [comment]
> @cagdas001, please resolve this ticket first #4875. The other is high priority.

@Oanh-and-only-Oanh its waiting for API Team

**Comment by cagdas001** [comment]
@Oanh-and-only-Oanh which branch I'll target for this one? Milestone is `Release by 10/8/2020` but there is no branch for it. I'm targeting `milestone-20200924` ?

**Comment by Oanh-and-only-Oanh** [comment]
Yes target https://github.com/topcoder-platform/community-app/tree/milestone-20200924
https://github.com/topcoder-platform/topcoder-react-lib/tree/milestone-20200924

**Comment by cagdas001** [comment]
It basically shows whatever it gets from resources-api in the `rating` field.

For e.g. https://api.topcoder.com/v5/resources?challengeId=457fe241-0687-4f60-a379-a3979a0d7210&roleId=732339e7-8e30-49d7-9198-cccf9451e221 You can take the member [Kriii](https://www.topcoder.com/members/kriii) for reference, as the issue exists & reproducible with this member.

resources-api gets the `rating` directly from member-api (maxRating): https://github.com/topcoder-platform/resources-api/blob/4edee0c7e666439e99e984c551f3753c447a9e0e/src/services/ResourceService.js#L148

If you're intending to show particular ratings for each track/challenge, I currently don't see such a response structure in both resources-api & member-api, there's just maxRating (in member-api) and nothing else.

So, either it should return something like the following, so we can get particular ratings

```json
ratings: {
  "SRM": 2878,
  "MARATHON_MATCH": 1791,
  ...
}
```
or should detect the challenge track/type based on`challengeId` and should get & return `rating` from member-api based on this. (however, as I said, I currently just see `maxRating` in `/members/<handle>` or `/members/<handle>/stats` response)

cc @Oanh-and-only-Oanh @rootelement @urwithat

**Comment by cagdas001** [comment]
Looks like `/members/<handle>/stats` returns all ratings. @rootelement If you get ratings from this endpoint you can access all

**Comment by cagdas001** [comment]
Ops. sorry for confusion. resource-api doesn't call member-api. directly queries from db https://github.com/topcoder-platform/resources-api/blob/4edee0c7e666439e99e984c551f3753c447a9e0e/src/common/helper.js#L154

**Comment by Oanh-and-only-Oanh** [comment]
@cagdas001, member registration tab shows MAX Rating out of all the rating member has and Submission tab shows MM Rating. They should not how different ratings. Both tabs should show MM Rating.

**Comment by cagdas001** [comment]
@Oanh-and-only-Oanh on my side, submissions tab also show wrong rating.

| Profile | Submissions Tab |
| ---- | ---- |
| ![chrome_2020-09-18_00-03-55](https://user-images.githubusercontent.com/28513447/93528236-b3ec7f80-f942-11ea-8a6b-a76cc9d839bd.png) | ![chrome_2020-09-18_00-04-48](https://user-images.githubusercontent.com/28513447/93528272-c1a20500-f942-11ea-8621-7a3c30949fc7.png) |

The issue is that, resource-api returns maxRating. It should work in one of the ways I explained in my comment (https://github.com/topcoder-platform/community-app/issues/4947#issuecomment-694458962) for us to show particular ratings:

> So, either it should return something like the following, so we can get particular ratings
> 
> ```json
> ratings: {
>   "SRM": 2878,
>   "MARATHON_MATCH": 1791,
>   ...
> }
> ```
> or should detect the challenge track/type based on`challengeId` and should return `rating` based on this.

**Comment by Oanh-and-only-Oanh** [comment]
@cagdas001 looks like MM ratings was fixed not long ago before it's now broken again. check out this closed ticket. https://github.com/topcoder-platform/community-app/issues/4730 Maybe it can help? Also @luizrrodrigues was the member who fixed it so maybe he can shed some light.

**Comment by cagdas001** [comment]
@Oanh-and-only-Oanh This issue has been closed without fixing the wrong rating issue: https://github.com/topcoder-platform/community-app/issues/4730#issuecomment-683772452

> @luizrrodrigues @sushilshinde , we are tracking this here #4794.
> 
> As the other issues in the ticket are fixed, we can go ahead.

**Comment by Oanh-and-only-Oanh** [comment]
@rootelement this a resource api issue and that it should return the maxRating instead of particular rating for the track. Please help.

**Comment by cagdas001** [comment]
It currently returns maxRating, we need particular ratings for different tracks/types. It should detect the challenge track/type by `challengeId` parameter, and return the `rating` based on this. Or should return all ratings so we can get particular rating at frontend. This comment explains the issue and what we need: https://github.com/topcoder-platform/community-app/issues/4947#issuecomment-694458962

**Comment by Oanh-and-only-Oanh** [comment]
@urwithat and @RishiRajSahu, can you please help with this ratings ticket?

**Comment by harshitmehta15** [comment]
Hi, @Oanh-and-only-Oanh Any updates on this? :) 

@ad-roc

**Comment by Oanh-and-only-Oanh** [comment]
Instead of resources API use member API. Update CA frontend to call different API.

**Comment by Oanh-and-only-Oanh** [comment]
@sushil, can you help with this issue during cool down?

**Comment by harshitmehta15** [comment]
@sushilshinde tldr; 
There are two ways to solve this
1) Use Member API (instead of Resource API) to fetch the specific rating based on what type of challenge is.
2) Update Resource API to return ratings of all the tracks.

**Comment by rootelement** [comment]
Resource api will not be updated to house that information. That's not relevant to the resource, and it's only there for basic convenience on the site. This has to be updated to pull that information from the member API. Consult with @urwithat on the appropriate usage.

**Comment by Oanh-and-only-Oanh** [comment]
Use the member api to fetch the specific rating based on what type of challenge is needed. The api exists and here is the curl for it:
Member v3 -
curl --location --request GET 'https://api.topcoder-dev.com/v3/members/tourist/stats'
Member v5 -
curl --location --request GET 'https://api.topcoder-dev.com/v5/members/stonefeang/stats'

**Comment by nursoltan-s** [comment]
@bug-bash-helper assign me

**Comment by nursoltan-s** [comment]
@Oanh-and-only-Oanh  
Hi, if it's open for pickup, can i take this issue?

**Comment by Oanh-and-only-Oanh** [comment]
hi @nursoltan-s, it's picked up by @rakibansary for now. thanks!

**Comment by nursoltan-s** [comment]
> hi @nursoltan-s, it's picked up by @rakibansary for now. thanks!

got it.

**Comment by rakibansary** [comment]
Fixed in https://github.com/topcoder-platform/topcoder-react-lib/pull/312
